### ISSUE TITLE
DeprecationWarning distutils and potential alternative

### ISSUE THREAD

**Issue by pannet1** [xonsh/xonsh#4613]
Whenever I open the xonsh I get error. I am using endeavourOS.
How to rectify this.
```
/usr/lib/python3.10/site-packages/xonsh/__amalgam__.py:19669: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
  rtn = self.loader.exec_module(module)

**Comment by anki-code** [comment]
> EndeavourOS is an Arch Linux based Linux distribution.

Weird. I'm on Arch but I have no issues.

**Comment by pannet1** [comment]
Thanks. I will try the usual fix, reinstall xonsh, though it makes no sense.

**Comment by anki-code** [comment]
@jnoortheen hi! May be you have some thoughts about this warning? I see [no distutils](https://github.com/xonsh/xonsh/search?q=distutils) in the source code.

**Comment by jnoortheen** [comment]
Yes I think it is related to the LooseVersion usage from distutils. @pannet1 could you post the xonfig info?

**Comment by pannet1** [comment]
```
@ xonfig                               {long_cmd_duration}
+------------------+------------------------+
| xonsh            | 0.11.0                 |
| Python           | 3.10.1                 |
| PLY              | 3.11                   |
| have readline    | True                   |
| prompt toolkit   | 3.0.24                 |
| shell type       | prompt_toolkit         |
| history backend  | json                   |
| pygments         | 2.10.0                 |
| on posix         | True                   |
| on linux         | True                   |
| distro           | endeavouros            |
| on wsl           | False                  |
| on darwin        | False                  |
| on windows       | False                  |
| on cygwin        | False                  |
| on msys2         | False                  |
| is superuser     | False                  |
| default encoding | utf-8                  |
| xonsh encoding   | utf-8                  |
| encoding errors  | surrogateescape        |
| on jupyter       | False                  |
| jupyter kernel   | None                   |
| xontrib 1        | abbrevs                |
| xontrib 2        | autovox                |
| xontrib 3        | autoxsh                |
| xontrib 4        | avox                   |
| xontrib 5        | coreutils              |
| xontrib 6        | distributed            |
| xontrib 7        | hist_navigator         |
| xontrib 8        | histcpy                |
| xontrib 9        | kitty                  |
| xontrib 10       | mpl                    |
| xontrib 11       | onepath                |
| xontrib 12       | output_search          |
| xontrib 13       | pdb                    |
| xontrib 14       | powerline3             |
| xontrib 15       | prompt_bar             |
| xontrib 16       | prompt_ret_code        |
| xontrib 17       | prompt_vi_mode         |
| xontrib 18       | readable-traceback     |
| xontrib 19       | ssh_agent              |
| xontrib 20       | vox                    |
| xontrib 21       | vox_tabcomplete        |
| xontrib 22       | voxapi                 |
| xontrib 23       | xlsd                   |
| xontrib 24       | z                      |
| RC file 1        | /home/pannet1/.xonshrc |
+------------------+------------------------+
```
> Yes I think it is related to the LooseVersion usage from distutils. @pannet1 could you post the xonfig info?

**Comment by jnoortheen** [comment]
I have added this change already. But I dont see any call to this function. It would be helpful to have the lines that generate this error. You can disable amalgamated import and get the actual file path eg. `env XONSH_NO_AMALGAMATE=1 xonsh`

**Comment by dyuri** [comment]
It might be my `ssh_agent` xontrib (and/or any other xontribs), I'm gonna fix that soon.

**Comment by dyuri** [comment]
New version (1.1.1) of `repassh` released, should fix the warning for python 3.10. (Have not tried all the other xontribs though.) @pannet1 please upgrade it and check if it fixes your issue.

**Comment by jnoortheen** [comment]
Thanks @dyuri  , closing the issue. @pannet1 please reopen if this issue stems from xonsh-master.
